### ISSUE TITLE
Multi-window tracing fails to write the last window raw file

### ISSUE THREAD

**Issue by bete0** [DynamoRIO/dynamorio#6245]
When trying to collect a multi-window trace to test PR #6165 , the raw trace for the last window fails to be to be written:
```
Hit tracing window #0 limit: disabling tracing.
Hit retrace threshold: enabling tracing for window #1.
Hit tracing window #1 limit: disabling tracing.
Hit retrace threshold: enabling tracing for window #2.
Hit tracing window #2 limit: disabling tracing.
Hit tracing window #10 limit: disabling tracing.
Hit retrace threshold: enabling tracing for window #11.
...
Hit retrace threshold: enabling tracing for window #23.
Hit tracing window #23 limit: disabling tracing.
Hit retrace threshold: enabling tracing for window #24.
Hit tracing window #24 limit: disabling tracing.
Fatal error: failed to create trace file ./drmemtrace.tool.drcacheoff.windows-timestamps.3949632.4419.dir/raw/window.0025/drmemtrace.tool.drcacheoff.windows-timestamps.3949632.6596.raw
```

**To Reproduce**
1. Checkout PR #6165 and build tests
2. Run the new `windows-timestamps` test

**Comment by derekbruening** [comment]
This seems to not be fully fixed as I hit what looks like the same window bug you hit where it tries to make a file in a non-existent dir at the very end:
```
Fatal error: failed to create trace file ./drmemtrace.simple_app.872162.7831.dir/raw/window.0002/drmemtrace.simple_app.872162.4788.raw.lz4

$ ls drmemtrace.simple_app.872162.7831.dir/raw
encodings.bin  funclist.log  modules.log  window.0000  window.0001
```

**Comment by derekbruening** [comment]
I'm also seeing this error in the windows-asm test:
```
WARNING: Input file for thread 1158195 is truncated
```
That means the trace is missing its footer.  It looks like PR #6165 modified the output matching to allow `.*` which hides this and doesn't fail the test, when it should fail as this is a problem.  Re-opening.

**Comment by derekbruening** [comment]
I am having trouble reproducing https://github.com/DynamoRIO/dynamorio/issues/6245#issuecomment-1701999696 but I have modified the fix from #6165 in a way that logically seems more correct for all cases, including non-split windows in the windows-asm test.  I also re-instated the original template for windows-asm.
